bosonprotocol/interface
# feat: replace Biconomy with internal meta-tx-gateway

## Description

- Core components dependencies are upgraded to use the internal meta-tx gateway instead of Biconomy.
- NOTE: Upgrade of core-components implies upgrade of Yup dependency version, meaning a few adaptation required (mostly type matching)
- In addition, the dApp now supports an env var to override the gatewayUrl. However this is optional (and not useful now the core-components is using the new gateway)
- The env var can be set differently on TESTING. STAGING and PRODUCTION deployment if needed. Again this is optional, as the default values from the core-components are the one to be used.
- The apiKeys and apiIds initially defined for Biconomy are still supported without requiring any change (everyone can keep their .env file as is)

## How to test

Check the network connection when submitting a transaction.
FYI, here are the new Urls, per env:
- testing: https://meta-tx-gateway-testing-114403180314.europe-west2.run.app
- staging: https://meta-tx-gateway-staging-114403180314.europe-west2.run.app
- production: https://meta-tx-gateway-114403180314.europe-west2.run.app

see more here: https://github.com/bosonprotocol/meta-tx-gateway/blob/main/docs/deployments.md
